src/observability/publisher.go

```diff
@@ -58,32 +58,40 @@ func ProcessSystemSummary() {
 
 func getAggregatedSummaryMap(tempSummarizerMap map[types.SystemSummary]types.SysSummaryTimeCount) map[types.SystemSummary]types.SysSummaryTimeCount {
 
-	var fileArr []string
+	fileArr := make(map[string][]string)
+	fileArrTmp := make(map[string][]common.SysPath)
+	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+
 	log.Info().Msgf("Events before aggregation: [%v]", len(tempSummarizerMap))
 
-	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
 	updatedSummarizerMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
 	for tss, tsstc := range tempSummarizerMap {
 		if tss.Operation == types.FileOperation {
-			fileArr = append(fileArr, tss.Destination)
+			destination := []string{}
+			destination = append(destination, fileArr[tss.PodName]...)
+			destination = append(destination, tss.Destination)
```

Comment:
@vishnusomank I do not understand this logic. Why do we add Podname and Destination both to same string slice ?
What is tss.Destination ?

Reply:
We are not adding pod_name to the string slice. We are creating a map[pod_name][]string to store the file paths of each pod before aggregating them. This will make the aggregation per pod and lead to more accurate summary information
cc: @wazir-ahmed